crates/types/src/message.rs

```diff
@@ -162,6 +162,12 @@ pub enum GeneralConsensusMessage<TYPES: NodeType> {
     /// Message with a quorum proposal.
     Proposal(Proposal<TYPES, QuorumProposal<TYPES>>),
 
+    /// A peer node needs a proposal from the leader.
+    ProposalRequested(TYPES::Time, TYPES::SignatureKey),
+
+    /// The leader has responded with a valid proposal.
+    LeaderProposalAvailable(Proposal<TYPES, QuorumProposal<TYPES>>),
```

Comment:
This message is needed to "finish" the request right? Is there any downside to sending-and-forgetting the request and getting a proposal, and then doing normal consensus operations with it?

Reply:
Yeah so basically it is the receiving node getting a `QuorumProposalResponseSend` event, and transforming it into a `QuorumProposalResponseRecv` event. This could be done any number of ways, I just opted to stratify things a little bit here because I prefer to have explicit events for each stage of the request chain. This GeneralConsensusMessage type honestly only exists to fulfill the need for there to be a consensus message that will not 
A. Be confused with a genuine proposal.
B. Be confused with a request for a proposal.

I'd happily name this anything else if there's any opinions there.

On your send-and-forget remark, I think generally I would like this, but it comes with two caveats given our async-focused design in HotShot

1. Right now, if I need to **guarantee** that data is available in my local in-memory state, I can *only* spin lock in a background thread (with some arbitrary delay).
2. I need a way to be able to halt a critical operation until my data has arrived.

As a result, the event dependency made the most sense. So, yes, *technically* we for sure could send and forget to get the proposal, the problem is we'd end up invariably waiting anyway, so by making the events explicitly say "I want a proposal right now for view x and I will wait until it is found" we can specialize what they're doing.